imports/client/ui/components/timeLine/TimeLine.jsx

```diff
@@ -42,64 +42,65 @@ export default class TimeLine extends React.Component {
 
   openMaxDatePicker = () => {
     this.refs.maxDatePicker.focus()
-    //console.log('maxDAte')
+
   }
-PauseOrClear = (event ) => {
 
+  pauseOrClear = (event ) => {
 
-   window.clearInterval(this.tid);
-  if (this.playOrResumeTrig == 1){
-    //if paused before :clean
-    if (this.PauseOrClearTrig ==1) {
-      k = Math.round(this.props.ui.minTime)
-      this.props.updateUI({currentSliderTime :  k })
-      this.PauseOrClearTrig= 0
-      this.playOrResumeTrig =0
-      }
-    else {
 
-      this.PauseOrClearTrig= 1
+     window.clearInterval(this.timerForPlay);
+    if (this.playOrResumeTrig == 1){
+      //if paused before :clean
+      if (this.pauseOrClearTrig ==1) {
+        k = Math.round(this.props.ui.minTime)
+        this.props.updateUI({currentSliderTime :  k })
+        this.pauseOrClearTrig= 0
+        this.playOrResumeTrig =0
+        }
+      else {
+
+        this.pauseOrClearTrig= 1
+        }
       }
+    else {
+      this.pauseOrClearTrig= 1
+      this.playOrResumeTrig =0
     }
-  else {
-    this.PauseOrClearTrig= 1
-    this.playOrResumeTrig =0
-  }
- }
+   }
 
-  PlayOrResume = (event ) => {
+  playOrResume = (event ) => {
     k = Math.round(this.props.ui.minTime)
 
-    if (!this.PauseOrClearTrig){
+    if (!this.pauseOrClearTrig){
       this.playOrResumeTrig = 0
-      this.PauseOrClearTrig =0
+      this.pauseOrClearTrig =0
       }
 
 
 
-    if (this.PauseOrClearTrig== 1) {
-      this.PauseOrClearTrig = 0;
+    if (this.pauseOrClearTrig== 1) {
+      this.pauseOrClearTrig = 0;
       k = this.props.ui.currentSliderTime
       }
 
     else {
-      if (this.tid){
-        window.clearInterval(this.tid);
+      if (this.timerForPlay){
+        window.clearInterval(this.timerForPlay);
         }
       }
 
     this.playOrResumeTrig = 1
     var seconds = parseInt((this.props.ui.maxTime-this.props.ui.minTime)/1000);
     var tempo = Math.floor(seconds);
 
-    this.tid = window.setInterval(function(){
+    this.timerForPlay = window.setInterval(function(){
```

Comment:
We don't want to bind any variables to the `window` element as it will make it global and can become a nightmare to debug. Instead we use React `state` to assign/cancel a timer, like explained in this post https://medium.com/@machadogj/timers-in-react-with-redux-apps-9a5a722162e8